### PR TITLE
Skip non virtual generic methods in build_imt_slots

### PR DESCRIPTION
**Pull request by bholmes**
When processing the methods of an interface, check that a generic method has the virtual flag to avoid corrupting the vtable.

### CODE REVIEW

File: src/mono/mono/metadata/object.c [hunk 1]
```diff
@@ -1576,7 +1578,7 @@ build_imt_slots (MonoClass *klass, MonoVTable *vt, gpointer* imt, GSList *extra_
 			for (method_slot_in_interface = 0; method_slot_in_interface < mcount; method_slot_in_interface++) {
 				MonoMethod *method = mono_class_get_method_by_index (iface, method_slot_in_interface);
 
-				if (method->is_generic)
+				if (method->is_generic && m_method_is_virtual(method))
 					has_generic_virtual = TRUE;
```
Comment (bholmes) [inline src/mono/mono/metadata/object.c:1582]: Pretty sure this is dead code, but I wanted to highlight the potential problem.

**Review by vargaz — APPROVED** [review]

**Comment by lambdageek** [comment]
I think this is preempted by #94437 - just need to try the test from this PR to validate that it's the same codepath.  Sorry it took us a while to revisit this.

File: src/mono/mono/metadata/object.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/metadata/object.c:1582]: yea, I ended up deleting this in #93349 - it was only used for COM interop

**Comment by lambdageek** [comment]
Yea this is the same issue.  It's been fixed by various backports in the January servicing releases (8.0.101, 7.0.15, 6.0.26)

Thanks for looking into it, @bholmed.

**Pull request closed without merging**